✨feat(chek-manager-role)권한 관련 설정 추가(NAILCASE-301)

hidongmin37 opened the pull request:
# Pull Request

[Contribution Guidelines](https://github.com/mobi-projects/nail-case-server/blob/main/WIKI/workflow/guidelines.md)

## PR 제목

✨feat(chek-manager-role)(NAILCASE-301) 권한 관련 설정 추가

## PR 설명

- manageronly 어노테이션 추가
- shop 엔티티에 체크 메소드 추가
- UseryType 제거

## 관련 이슈

- resolves: [NAILCASE-301](https://nailcase.atlassian.net/browse/NAILCASE-301))

## 체크리스트

- [x] 코드가 올바르게 동작하는지 확인했습니다.
- [x] 관련 문서가 없습니다.
- [ ] 관련 테스트를 추가했습니다. / 테스트가 필요 없는 PR입니다.

## 스크린샷 (선택사항)

변경 사항과 관련된 스크린샷을 여기에 추가하세요.

## 추가 정보 (선택사항)

여기에 PR에 대한 추가 정보를 기입하세요.


[NAILCASE-301]: https://nailcase.atlassian.net/browse/NAILCASE-301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

hidongmin37 committed:
✨feat(chek-manager-role)(NAILCASE-301)

- manageronly 어노테이션 추가
- shop 엔티티에 체크 메소드 추가
- UseryType 제거

resolves: [NAILCASE-301](https://nailcase.atlassian.net/browse/NAILCASE-301))

qushe8r commented on src/main/java/com/nailcase/model/dto/UserPrincipal.java:
```diff
@@ -1,11 +1,19 @@
+import com.nailcase.model.enums.Role;
 
 public interface UserPrincipal {
 	Long getId();
 
 	String getEmail();
 
-	UserType getUserType();
+	Role getRole();
+
+	default boolean isMember() {
+		return this instanceof MemberDetails;
+	}
+
+	default boolean isNailArtist() {
+		return this instanceof NailArtistDetails;
+	}
 }
```
sealed interface를 고려해봐도 되지 않을까요?
MemberDetails나 NailArtistDetails만 사용해야 하는 인터페이스 인거 같아서요.
저도 sealed 를 써본적은 없는데 상속을 제한하는 거라고만 알고있어서 나중에 공부해 보면 좋을 것 같습니다.

hidongmin37 replied:
오호 한번 알아보겠습니다

qushe8r reviewed (approved):
수고하셨습니다.

hidongmin37 committed:
✨feat(chek-manager-role)(NAILCASE-301)

- sealed interface를 활용하여 userDetails 상속할 수 있는 클래스 제한

resolves: [NAILCASE-301](https://nailcase.atlassian.net/browse/NAILCASE-301))

hidongmin37 committed:
✨feat(chek-manager-role)(NAILCASE-301)

- sealed interface를 활용하여 userDetails 상속할 수 있는 클래스 제한 후 수정

resolves: [NAILCASE-301](https://nailcase.atlassian.net/browse/NAILCASE-301))

nickkies reviewed (approved):
고생하셧습니다! 

그런데 테스트코드 쪽에는 반영이 덜 된거같은데 빌드할때 테스트 통과 되나요??

hidongmin37 commented:
> 고생하셧습니다!
> 
> 그런데 테스트코드 쪽에는 반영이 덜 된거같은데 빌드할때 테스트 통과 되나요??

아하 로컬에서 실행할때 됐는데, 날리고 재빌드하니깐 에러나네요 그부분 수정해서 머지하도록 하겠습니다!

hidongmin37 committed:
✨feat(chek-manager-role)(NAILCASE-301)

- test코드 수정

resolves: [NAILCASE-301](https://nailcase.atlassian.net/browse/NAILCASE-301))
